cmd/pod-scaler/admission.go

```diff
@@ -273,36 +273,34 @@ func preventUnschedulable(resources *corev1.ResourceRequirements, cpuCap int64,
-	for i := range pod.Spec.InitContainers {
-		meta := pod_scaler.MetadataFor(pod.ObjectMeta.Labels, pod.ObjectMeta.Name, pod.Spec.InitContainers[i].Name)
-		resources, recommendationExists := server.recommendedRequestFor(meta)
-		if recommendationExists {
-			logger.Debugf("recommendation exists for: %s", pod.Spec.InitContainers[i].Name)
-			useOursIfLarger(&resources, &pod.Spec.InitContainers[i].Resources, determineName(pod.Labels, pod.Name, pod.Spec.InitContainers[i].Name), reporter, logger)
-			if mutateResourceLimits {
-				reconcileLimits(&pod.Spec.InitContainers[i].Resources)
-			}
-		}
-		preventUnschedulable(&pod.Spec.InitContainers[i].Resources, cpuCap, memoryCap, logger)
-	}
-	for i := range pod.Spec.Containers {
-		meta := pod_scaler.MetadataFor(pod.ObjectMeta.Labels, pod.ObjectMeta.Name, pod.Spec.Containers[i].Name)
-		resources, recommendationExists := server.recommendedRequestFor(meta)
-		if recommendationExists {
-			logger.Debugf("recommendation exists for: %s", pod.Spec.Containers[i].Name)
-			useOursIfLarger(&resources, &pod.Spec.Containers[i].Resources, determineName(pod.Labels, pod.Name, pod.Spec.Containers[i].Name), reporter, logger)
-			if mutateResourceLimits {
-				reconcileLimits(&pod.Spec.Containers[i].Resources)
+	mutateResources := func(containers []corev1.Container) {
+		for i := range containers {
+			meta := pod_scaler.MetadataFor(pod.ObjectMeta.Labels, pod.ObjectMeta.Name, containers[i].Name)
+			resources, recommendationExists := server.recommendedRequestFor(meta)
+			if recommendationExists {
+				logger.Debugf("recommendation exists for: %s", containers[i].Name)
+				useOursIfLarger(&resources, &containers[i].Resources, fmt.Sprintf("%s-%s", pod.Name, containers[i].Name), determineWorkloadType(pod.Annotations, pod.Labels), reporter, logger)
+				if mutateResourceLimits {
+					reconcileLimits(&containers[i].Resources)
+				}
 			}
+			preventUnschedulable(&containers[i].Resources, cpuCap, memoryCap, logger)
 		}
-		preventUnschedulable(&pod.Spec.Containers[i].Resources, cpuCap, memoryCap, logger)
 	}
+	mutateResources(pod.Spec.InitContainers)
+	mutateResources(pod.Spec.Containers)
 }
 
-// determineName returns the string that will be used in Prometheus metrics to identify the workload
-func determineName(podLabels map[string]string, podName, containerName string) string {
-	if value, exists := podLabels["prow.k8s.io/job"]; exists {
-		return value
+// determineWorkloadType returns the workflow type
+func determineWorkloadType(annotations, labels map[string]string) string {
+	if _, isBuildPod := annotations[buildv1.BuildLabel]; isBuildPod {
+		return "build"
+	}
+	if _, isProwjob := labels["prow.k8s.io/job"]; isProwjob {
+		return "prowjob"
+	}
+	if _, isRehearsal := labels[rehearse.Label]; isRehearsal {
+		return "rehearsal"
 	}
-	return fmt.Sprintf("%s-%s", podName, containerName)
+	return "undefined"
```

Comment:
not sure about "undefined", but I didn't want to return an empty string

Reply:
I think that is fine for now.